compiler/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -48,12 +48,17 @@ pub enum TypeCheckError {
     TypeMismatchWithSource { expected: Type, actual: Type, span: Span, source: Source },
     #[error("Expected type {expected_kind:?} is not the same as {expr_kind:?}")]
     TypeKindMismatch { expected_kind: String, expr_kind: String, expr_span: Span },
+    // TODO(https://github.com/noir-lang/noir/issues/6238): implement handling for larger types
+    #[error("Expected type {expected_kind:?} when evaluating globals: found {expr_kind:?}, size checks are being implemented for this case")]
+    EvaluatedGlobalPartialSizeChecks { expected_kind: String, expr_kind: String, expr_span: Span },
     #[error("Expected {expected:?} found {found:?}")]
     ArityMisMatch { expected: usize, found: usize, span: Span },
     #[error("Return type in a function cannot be public")]
     PublicReturnType { typ: Type, span: Span },
     #[error("Cannot cast type {from}, 'as' is only for primitive field or integer types")]
-    InvalidCast { from: Type, span: Span },
+    InvalidCast { from: Type, span: Span, reason: String },
+    #[error("Casting value with {from} to a smaller type")]
```

Comment:
"Casting value with i32 to a smaller type" reads a bit awkwardly

Reply:
Updated to `Casting value with type {from} to a smaller type ({to})`: ok?

Reply:
How about "Casting value of type {from} to a smaller type ({to})"

"casting value with type" sometimes reads to me like you're casting to the from type.